src/test/ui/macros/macro-backtrace-invalid-internals.rs

```diff
@@ -48,13 +48,13 @@ macro_rules! fake_anon_field_expr {
 
 macro_rules! real_method_stmt {
      () => {
-          2.0.powi(2) //~ ERROR can't call method `powi` on ambiguous numeric type `{float}`
+          2.0.recip() //~ ERROR can't call method `recip` on ambiguous numeric type `{float}`
```

Comment:
How come these changes ended up being necessary?

Reply:
At least one of the modified tests is `#![no_std]` and `powi` has been removed form libcore as discussed above. However this one isn’t and if I revert this diff chuck the error message is ``error[E0599]: no method named `powi` found for type `{float}` in the current scope`` instead of the expected ``error[E0689]: can't call method `powi` on ambiguous numeric type `{float}` ``. I don’t really understand why right now, I’ll need to investigate…

Reply:
Ah ok that makes sense. I think it's ok to not worry much about the error here, it probably has to do with trait vs inherent dispatch and doesn't seem to indicate a bug either way.

Reply:
After much squinting at rustc code, I found out that not only E0599 only looks at traits and not inherent impls to see if the method exists (which explains this change), but it doesn’t even check if those traits are are implemented for any relevant type. And only error messages are affected, not actual name resolution, so leaving this as is is less worrying that I thought yesterday.